c2/c2-client-bundle/c2-client-service/src/test/java/org/apache/nifi/c2/client/service/operation/DebugOperationHandlerTest.java

```diff
@@ -0,0 +1,234 @@
+import static org.apache.nifi.c2.protocol.api.C2OperationState.OperationState.FULLY_APPLIED;
+import static org.apache.nifi.c2.protocol.api.C2OperationState.OperationState.NOT_APPLIED;
+import static org.apache.nifi.c2.protocol.api.OperandType.DEBUG;
+import static org.apache.nifi.c2.protocol.api.OperationType.TRANSFER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.nifi.c2.client.api.C2Client;
+import org.apache.nifi.c2.protocol.api.C2Operation;
+import org.apache.nifi.c2.protocol.api.C2OperationAck;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class DebugOperationHandlerTest {
+
+    private static final String OPERATION_ID = "operationId";
+    private static final String C2_DEBUG_UPLOAD_ENDPOINT = "https://host/c2/api/upload";
+    private static final String DEFAULT_FILE_CONTENT = "some_textual_data";
+    private static final List<Path> VALID_BUNDLE_FILE_LIST = singletonList(Paths.get("path_to_file"));
+    private static final Predicate<String> DEFAULT_CONTENT_FILTER = text -> true;
+
+    @Mock
+    private C2Client c2Client;
+
+    @TempDir
+    private File tempDir;
+
+    private static Stream<Arguments> invalidConstructorArguments() {
+        C2Client mockC2Client = mock(C2Client.class);
+        return Stream.of(
+            Arguments.of(null, null, null),
+            Arguments.of(null, VALID_BUNDLE_FILE_LIST, DEFAULT_CONTENT_FILTER),
+            Arguments.of(mockC2Client, null, DEFAULT_CONTENT_FILTER),
+            Arguments.of(mockC2Client, emptyList(), DEFAULT_CONTENT_FILTER),
+            Arguments.of(mockC2Client, VALID_BUNDLE_FILE_LIST, null)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("invalidConstructorArguments")
```

Comment:
Very nice and thorough test just one note, you could customise this with referencing the parameters so when it is executed it is printed what kind of parameter combination is passed for each test, it is helpful to instantly see what went wrong in case of errors.

Reply:
Added parameter reference